plugins/discourse-details/assets/javascripts/initializers/apply-details.js

```diff
@@ -6,26 +6,19 @@ function initializeDetails(api) {
     id: "discourse-details",
   });
 
-  api.addToolbarPopupMenuOptionsCallback(() => {
+  api.addComposerToolbarPopupMenuOption(() => {
```

Comment:
Any reason for this to still pass a function? Could it be simplified to something like this?

```js
api.addComposerToolbarPopupMenuOption({
  action(toolbarEvent){
     toolbarEvent.applySurround(
        "\n" + `[details="${I18n.t("composer.details_title")}"]` + "\n",
        "\n[/details]\n",
        "details_text",
        { multiline: false }
      );
  },
  icon: "caret-right",
  label: "details.title",
});
```